docs/source/tutorial/train_loop.rst

```diff
@@ -0,0 +1,328 @@
+    5. Run the optimizer to update those parameters.
+    6. (Optional): Check the network performance on the validation/test set.
+
+1. Prepare the dataset
+''''''''''''''''''''''
+
+Chainer contains some built-in functions to use some popular datasets like MNIST, CIFAR10/100, etc. Those can automatically download the data from servers and provide dataset objects which are easy to use.
+
+The below code shows how to retrieve the MNIST dataset from the server and save an image from its training split to make sure the images are correctly obtained.
+
+.. testcode::
+
+    import matplotlib.pyplot as plt
+    from chainer.datasets import mnist
+
+    # Download the MNIST data if you haven't downloaded it yet
+    train, test = mnist.get_mnist(withlabel=True, ndim=1)
+
+    # Display an example from the MNIST dataset.
+    # `x` contains the input image array and `t` contains that target class
+    # label as an integer.
+    x, t = train[0]
+    plt.imshow(x.reshape(28, 28), cmap='gray')
+    plt.savefig('5.png')
+    print('label:', t)
+
+.. testoutput::
+
+    label: 5
+
+The saved image ``5.png`` will look like:
+
+.. image:: ../../image/train_loop/5.png
+
+
+2. Create the dataset iterators
+'''''''''''''''''''''''''''''''
+
+Although this is an optional step, we'd like to introduce the :class:`~chainer.dataset.Iterator` class that retrieves a set of data and labels from the given dataset to easily make a mini-batch. There are some subclasses that can perform the same thing in different ways, e.g., using multi-processing to parallelize the data loading part, etc.
+
+Here, we use :class:`~chainer.iterators.SerialIterator`, which is also a subclass of :class:`~chainer.dataset.Iterator` in the below example code. The :class:`~chainer.iterators.SerialIterator` can provide mini-batches with or without shuffling the order of data in the given dataset.
+
+All :class:`~chainer.dataset.Iterator` s produce a new mini-batch by calling its :meth:`~chainer.dataset.Iterator.next` method. All
+:class:`~chainer.dataset.Iterator` s also have properties to know how many times we have taken all the data from the given dataset (:attr:`~chainer.dataset.Iterator.epoch`) and whether the next mini-batch will be the start of a new epoch (:attr:`~chainer.dataset.Iterator.is_new_epoch`), and so on.
+
+The below code shows how to create a :class:`~chainer.iterators.SerialIterator` object from a dataset object.
+
+.. testcode::
+
+    from chainer import iterators
+
+    # Choose the minibatch size.
+    batchsize = 128
+
+    train_iter = iterators.SerialIterator(train, batchsize)
+    test_iter = iterators.SerialIterator(test, batchsize,
+                                         repeat=False, shuffle=False)
+
+.. note::
+
+    :class:`~chainer.dataset.iterator` s can take a built-in Python list as a given dataset. It means that the below code example is able to work,
```

Comment:
- Ditto (`\s`)
- Ditto ("*the below code example*" is not grammatically correct)